packages/framework-provider-azure/src/helpers/query-helper.ts

```diff
@@ -314,6 +317,57 @@ function buildProjections(projections: ProjectionFor<unknown> | string = '*'): s
     .join(', ')
 }
 
+function preprocessProjections(projections: ProjectionFor<unknown>): ProjectionFor<unknown> {
+  const processed = new Set<string>()
+
+  Object.values(projections).forEach((field: string) => {
+    const parts = field.split('.')
+    const arrayIndices = parts.reduce((acc, part, index) => {
+      if (part.endsWith('[]')) acc.push(index)
+      return acc
+    }, [] as number[])
+
+    if (arrayIndices.length === 0) {
+      // Case 1: No arrays in the projection
+      processed.add(field)
+    } else if (arrayIndices[0] === 0) {
+      // Case 2.1 and 2.2: Array at the top level
+      if (arrayIndices.length === 1) {
+        // Case 2.1: Top-level array not followed by another array
+        processed.add(field)
+      } else {
+        // Case 2.2: Top-level array followed by a subarray
+        const processedField = parts.slice(0, arrayIndices[1] + 1).join('.')
+        processed.add(processedField.slice(0, -2)) // Remove '[]' from the last part
+      }
+    } else if (arrayIndices[0] === 1) {
+      // Case 2.3: Array nested within a top-level property
+      if (arrayIndices.length === 1) {
+        // Case 2.3.1: No arrays follow in the nesting
+        processed.add(field)
+      } else {
+        // Case 2.3.2: Nested arrays after it
+        const processedField = parts.slice(0, arrayIndices[1] + 1).join('.')
+        processed.add(processedField.slice(0, -2)) // Remove '[]' from the last part
+      }
+    } else {
+      // Case 2.4: Array nested deeper than a top-level property
+      const processedField = parts.slice(0, arrayIndices[0] + 1).join('.')
+      processed.add(processedField.slice(0, -2)) // Remove '[]' from the last part
+    }
```

Comment:
I'm not sure but the three `else` blocks for cases 2.2, 2.3.2 and 2.4 are repeated. Maybe if we swap the nested conditions and put in line 333 the condition `else if (arrayIndices.length === 1)` instead, we could save some repetition of code

Reply:
This code is a bit hard to read. Question: Why `arrayIndices[0] === 1` means "Array nested within a top-level property"?

Reply:
Thanks for the suggestion. Applied in commit 6813d46.
Initially, it was this lengthy because it was easier to follow which cases apply.